src/app/components/algorithms/implementation-view/implementation-nisq-analyzer-qpu-selection/implementation-nisq-analyzer-qpu-selection.component.ts

```diff
@@ -268,11 +344,209 @@ export class ImplementationNisqAnalyzerQpuSelectionComponent implements OnInit {
       });
   }
 
+  prioritize(): void {
+    this.rankings = [];
+    this.utilService
+      .createDialog(
+        ImplementationNisqAnalyzerQpuSelectionPrioritizationDialogComponent,
+        {
+          title: 'Prioritize QPU-Selection-Analysis',
+        }
+      )
+      .afterClosed()
+      .subscribe((dialogResult) => {
+        let totalSum = 0;
+        if (dialogResult) {
+          this.loadingMCDAJob = true;
+          this.prioritizationJob = undefined;
+          this.prioritizationJobReady = false;
+          // calculate SMART with new assigned points
+          dialogResult.criteriaAndValues.forEach((obj) => {
+            totalSum += obj.points;
+          });
+          dialogResult.criteriaAndValues.forEach((obj) => {
+            if (obj.points !== 0) {
+              obj.weight = obj.points / totalSum;
+            } else {
+              obj.weight = 0;
+            }
+          });
+          let numberOfCriterion = 0;
+          dialogResult.criteriaAndValues.forEach((obj) => {
+            const criterionValue: CriterionValue = {
+              description: { title: 'points', subTitle: obj.points.toString() },
+              criterionID: obj.id,
+              valueOrValues: [{ real: obj.weight }],
+              mcdaMethod: dialogResult.mcdaMethod,
+            };
+            this.mcdaService
+              .updateCriterionValue({
+                methodName: dialogResult.mcdaMethod,
+                criterionId: obj.id,
+                body: criterionValue,
+              })
+              .subscribe(
+                () => {
+                  numberOfCriterion++;
+                  if (
+                    numberOfCriterion === dialogResult.criteriaAndValues.length
+                  ) {
+                    this.mcdaService
+                      .prioritizeCompiledCircuitsOfJob({
+                        methodName: dialogResult.mcdaMethod,
+                        jobId: this.analyzerJob.id,
+                      })
+                      .subscribe((job) => {
+                        this.prioritizationJob = job;
+                        this.prioritizationJobReady = job.ready;
+
+                        this.utilService.callSnackBar(
+                          'Successfully created prioritization job "' +
+                            job.id +
+                            '".'
+                        );
+
+                        this.pollingAnalysisJobData = interval(2000)
```

Comment:
You can extract this into a separate method. Then this method becomes shorter and easier to understand.

Reply:
Done